# HG changeset patch
# Parent 44f263efd1a56ee459bfdb0cc5f66484bf3bdb85
# User Frédéric Wang <fred.wang@free.fr>
munderover: take into account the base child vertical metrics for the computation of the munderover ascent/descent.

diff --git a/layout/mathml/nsMathMLmunderoverFrame.cpp b/layout/mathml/nsMathMLmunderoverFrame.cpp
--- a/layout/mathml/nsMathMLmunderoverFrame.cpp
+++ b/layout/mathml/nsMathMLmunderoverFrame.cpp
@@ -431,16 +431,17 @@ nsMathMLmunderoverFrame::Place(nsRenderi
   // from where to attach the underscript. Hence if the underscript is empty,
   // we should end up like <mover>. If the overscript is empty, we should
   // end up like <munder>.
 
   nsBoundingMetrics bmAnonymousBase = mBoundingMetrics;
   nscoord ascentAnonymousBase =
     NS_MAX(mBoundingMetrics.ascent + overDelta2,
            overSize.ascent + bmOver.descent + overDelta1 + bmBase.ascent);
+  ascentAnonymousBase = NS_MAX(ascentAnonymousBase, baseSize.ascent);
 
   GetItalicCorrection(bmAnonymousBase, correction);
 
   // Width of non-spacing marks is zero so use left and right bearing.
   nscoord underWidth = bmUnder.width;
   if (!underWidth) {
     underWidth = bmUnder.rightBearing - bmUnder.leftBearing;
     dxUnder = -bmUnder.leftBearing;
@@ -470,16 +471,19 @@ nsMathMLmunderoverFrame::Place(nsRenderi
   mBoundingMetrics.rightBearing = 
     NS_MAX(dxAnonymousBase + bmAnonymousBase.rightBearing, dxUnder + bmUnder.rightBearing);
 
   aDesiredSize.ascent = ascentAnonymousBase;
   aDesiredSize.height = aDesiredSize.ascent +
     NS_MAX(mBoundingMetrics.descent + underDelta2,
            bmAnonymousBase.descent + underDelta1 + bmUnder.ascent +
              underSize.height - underSize.ascent);
+  aDesiredSize.height = NS_MAX(aDesiredSize.height,
+                               aDesiredSize.ascent +
+                               baseSize.height - baseSize.ascent);
   aDesiredSize.width = mBoundingMetrics.width;
   aDesiredSize.mBoundingMetrics = mBoundingMetrics;
 
   mReference.x = 0;
   mReference.y = aDesiredSize.ascent;
 
   if (aPlaceOrigin) {
     nscoord dy;
